inst/include/Rcpp/String.h

```diff
@@ -64,10 +63,10 @@ namespace Rcpp {
             RCPP_STRING_DEBUG("String(const String&)");
         }
 
-        String(const String& other, const std::string& enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
+        String(const String& other, cetype_t enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
```

Comment:
Actually this is mostly to make our ctor matrix complete. These two ctors can be removed. If we want to change the encoding of `String s`, we can just `s.set_encoding()`. `set_encoding` is public.